validator/src/machine/keygen/confirmed.ts

```diff
@@ -0,0 +1,82 @@
+import type { KeyGenClient } from "../../consensus/keyGen/client.js";
+import type { ProtocolAction } from "../../consensus/protocol/types.js";
+import { keyGenConfirmedEventSchema } from "../../consensus/schemas.js";
+import type { SigningClient } from "../../consensus/signing/client.js";
+import type { ConsensusDiff, ConsensusState, MachineStates, StateDiff } from "../types.js";
+
+export const handleKeyGenConfirmed = async (
+	keyGenClient: KeyGenClient,
+	signingClient: SigningClient,
+	consensusState: ConsensusState,
+	machineStates: MachineStates,
+	eventArgs: unknown,
+	logger?: (msg: unknown) => void,
+): Promise<StateDiff> => {
+	// A participant has confirmed their participation in the key gen ceremony
+	// Ignore if not in "collecting_confirmations" state
+	if (machineStates.rollover.id !== "collecting_confirmations") {
```

Comment:
Just adding a note here, but I don't think this is correct: if we have 3 participants where 1 and 2 honestly share secret shares and 3 dishonestly shares with 2; then 1 and 3 can enter the `collecting_confirmations` state, but 2 cannot because it is missing a share resulting from the complaint system. It needs to additionally wait for a `KeyGenComplaintResponse` with the missing share before it can construct its signing share. This means that we need to potentially allow confirmations while waiting for pending complaints.

I would add a note here, as it is something to consider for when we implement complaints in the validator (so not for this PR).